proto/pulumi/errors.proto

```diff
@@ -23,3 +23,18 @@ message ErrorCause {
     string stackTrace = 2;
 }
 
+// An error that can be returned from a component provider and includes details of the
+// error, which can be multiple properties.
+message InvalidInputPropertiesError {
+    // A single invalid input property.
+    message InvalidInputPropertyError {
+        // The key to the property that is invalid.
```

Comment:
Every key _is_ a path, you only need path here that will allow just "foo" but also "foo.bar".